### PR TITLE
Added Factions to Voiced.

### PR DESCRIPTION
**Pull request by Mailaender**
Required for Red Alert 2 which has complicated rules for faction specific voices.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Voiced.cs [hunk 1]
```diff
@@ -19,6 +20,9 @@ public class VoicedInfo : ITraitInfo
 		[Desc("Which voice set to use.")]
 		[VoiceSetReference] public readonly string VoiceSet = null;
 
+		[Desc("Which faction voice to use.")]
+		public readonly string[] Factions = null;
```
Comment (Phrohdoh) [inline OpenRA.Mods.Common/Traits/Voiced.cs]: Arrays need to be empty, not null, by default.

**Comment by pchote** [comment]
This conflicts with the plan to extract all the ugly mod-specific sound quirks into "sound sequences".  Are you able to live with an ra2-mod specific `RA2Voiced` trait subclass until this gets solved more generally upstream?

**Comment by pchote** [comment]
You should be able to solve your engineer voices using our existing faction-specific voices support.  Look at how d2k does it.

**Comment by Mailaender** [comment]
Been there, tried that. https://github.com/OpenRA/ra2/pull/49

**Comment by GraionDilach** [comment]
:+1: for the `RA2Voiced` trait. This is still overlapping with the faction prefixes in the voice.yaml and such hackish approaches should be downstream.

**Comment by abcdefg30** [comment]
Why is that hacky? To me this Looks valid and ok.

**Comment by Iamgoofball** [comment]
so whats the replacement for this PR because we kind of need this for RA2

**Comment by Mailaender** [comment]
We can either implement a `FactionVoiced` trait in the OpenRA.Mods.RA2.dll or go with something like https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.Mods.Common/Graphics/TilesetSpecificSpriteSequence.cs#L70 but for voices. For the RA2 soviet engineer we will just need to make sure that it retries another one if it can't find a variant.

**Pull request closed without merging**